python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -338,9 +359,16 @@ def __init__(
         self._asset_selection = check.opt_inst_param(
             asset_selection, "asset_selection", AssetSelection
         )
+        resource_arg_names: Set[str] = {arg.name for arg in get_resource_args(self._raw_fn)}
+
+        self._required_resource_keys = (
+            check.opt_set_param(required_resource_keys, "required_resource_keys", of_type=str)
+            | resource_arg_names
+        )
```

Comment:
Is this the behavior in our other decorators? The union of the parameters and the `required_resourc_keys` arguments?